packages/design/src/DataTable/Paged/Pager/Pager.jsx

```diff
@@ -21,17 +21,45 @@ import { Text, Flex } from 'design';
 import PropTypes from 'prop-types';
 
 export default function Pager(props) {
-  const { startFrom = 0, endAt = 0, totalRows = 0, onPrev, onNext } = props;
+  const {
+    startFrom = 0,
+    endAt = 0,
+    totalRows = 0,
+    onPrev,
+    onNext,
+    moreEvents,
+  } = props;
   const isPrevDisabled = totalRows === 0 || startFrom === 0;
   const isNextDisabled = totalRows === 0 || endAt === totalRows;
   const initialStartFrom = totalRows > 0 ? startFrom + 1 : 0;
 
+  let $showingTotalEvents = (
+    <Text typography="body2" color="primary.contrastText">
+      SHOWING <strong>{initialStartFrom}</strong> - <strong>{endAt}</strong> of{' '}
+      <strong>{totalRows}</strong>
+    </Text>
+  );
+
+  if (moreEvents) {
+    $showingTotalEvents = (
+      <Flex alignItems="center">
+        <Text typography="body2" color="primary.contrastText">
+          SHOWING <strong>{initialStartFrom}</strong> - <strong>{endAt}</strong>{' '}
+          of <strong>{totalRows}</strong>
+        </Text>
+        <StyledButtonLink
+          onClick={moreEvents.fetch}
+          disabled={moreEvents.disabled}
+        >
+          Fetch More
+        </StyledButtonLink>
+      </Flex>
+    );
+  }
+
```

Comment:
Not sure I understand the reason for duplicating the paragraph but below looks simpler

```tsx
export default function Pager(props) {
  const {
    startFrom = 0,
    endAt = 0,
    totalRows = 0,
    onPrev,
    onNext,
    onFetch,
    fetchStatus,
  } = props;

  const isFetchDisabled = fetchStatus !== 'disabled';
  const isFetching = fetchStatus === 'loading';
  const isPrevDisabled = totalRows === 0 || startFrom === 0;
  const isNextDisabled = totalRows === 0 || endAt === totalRows;
  const initialStartFrom = totalRows > 0 ? startFrom + 1 : 0;

  return (
    <>
      <Flex alignItems="center">
        <Text typography="body2" color="primary.contrastText">
          SHOWING <strong>{initialStartFrom}</strong> - <strong>{endAt}</strong>{' '}
          of <strong>{totalRows}</strong>
        </Text>
        {isFetchDisabled && (
          <StyledButtonLink onClick={onFetch} disabled={isFetching}>
            Fetch More
          </StyledButtonLink>
        )}
      </Flex>
      <StyledButtons>
        <button
          onClick={onPrev}
          title="Previous Page"
          disabled={isPrevDisabled}
        >
          <CircleArrowLeft fontSize="3" />
        </button>
        <button onClick={onNext} title="Next Page" disabled={isNextDisabled}>
          <CircleArrowRight fontSize="3" />
        </button>
      </StyledButtons>
    </>
  );
}
```

Reply:
revised